File: Hash Table/1805.Number of Different Integers in a String/Solution.py
class Solution:
    def numDifferentIntegers(self, word: str) -> int:
        s = set()
        ans = 0
        cur, word = "", word + "a"
        for c in word:
            if '0' <= c <= '9':
                cur += c
            else:
                if not cur:
                    continue
                v = int(cur)
                if v not in s:
                    ans += 1
                s.add(v)
                cur = ""
        return ans

File: Hash Table/1805.Number of Different Integers in a String/Solution.java
class Solution {
    public int numDifferentIntegers(String word) {
        int ans = 0;
        StringBuilder cur = new StringBuilder();
        Set<String> set = new HashSet<>();
        word += "a";
        for (int i = 0; i < word.length(); i++) {
            if (Character.isDigit(word.charAt(i))) {
                if (cur.length() == 1 && "0".equals(cur.toString())) {
                    cur = new StringBuilder();
                }
                cur.append(word.charAt(i));
            } else {
                if (cur.length() > 0) {
                    if (!set.contains(cur.toString())) {
                        ans++;
                        set.add(cur.toString());
                    }
                    cur = new StringBuilder();
                }
            }
        }
        return ans;
    }
}
